test/gurka/test_build_admin.cc

```diff
@@ -0,0 +1,128 @@
+protected:
+  static gurka::map admin_map;
+
+  static void SetUpTestSuite() {
+    constexpr double gridsize = 10;
+
+    const std::string ascii_map = R"(
+          A-------B-------C
+          |       |       |
+          |       |       |
+          |       |       |
+          F-------E-------D
+    )";
+
+    // To define an administrative boundary, the nodes must form a closed polygon.
+    const gurka::ways ways = {{"ABCDEFA",
+                               {
+                                   {"highway", "primary"},
+                               }},
+                              {"ABEFA",
+                               {
+                                   {"highway", "primary"},
+                               }},
+                              {"BCDEB",
+                               {
+                                   {"highway", "primary"},
+                               }}};
+
+    const gurka::relations relations = {{{{{gurka::way_member, "ABEFA", "outer"}}},
+                                         {
+                                             {"type", "boundary"},
+                                             {"boundary", "administrative"},
+                                             {"admin_level", "4"},
+                                         }},
+                                        {
+                                            {{{gurka::way_member, "BCDEB", "outer"}}},
+                                            {
+                                                {"type", "boundary"},
+                                                {"boundary", "administrative"},
+                                                {"admin_level", "4"},
+                                            },
+                                        },
+                                        {{{{gurka::way_member, "ABCDEFA", "outer"}}},
+                                         {
+                                             {"type", "boundary"},
+                                             {"boundary", "administrative"},
+                                             {"admin_level", "2"},
+                                         }}};
+
+    const auto layout = gurka::detail::map_to_coordinates(ascii_map, gridsize);
+    admin_map = gurka::buildtiles(layout, ways, {}, relations, "test/data/admin");
+  }
+};
+
+gurka::map AdminTest::admin_map;
+
+TEST_F(AdminTest, test) {
+  // create a config file
+  const std::string config_file = "test/data/admin/config";
+  std::ofstream file;
+  file.open(config_file, std::ios_base::trunc);
+  ASSERT_TRUE(file.is_open());
+  file << "{ \
+    \"mjolnir\": { \
+    \"concurrency\": 1, \
+    \"id_table_size\": 1000, \
+    \"tile_dir\": \"test/data/admin\", \
+    \"admin\": \"test/data/admin.sqlite\", \
+    \"timezone\": \"test/data/not_needed.sqlite\" \
+    } \
+  }";
+  file.close();
+
+  boost::property_tree::ptree pt;
+  rapidjson::read_json(config_file, pt);
```

Comment:
you can create the config directly without writing a file, which i would recommend. it looks lsomething like th is:

```
boost::property_tree::ptree pt;
pt.put("mjolnir.concurrency", 1);
pt.put("mjolnir.tile_dir", "foo/bar");
...
```

Reply:
That would look cleaner, yes. I will do that. (I was sort-of modeling my code after other unit tests, we may want to apply your suggestion to those tests, too, eventually)